### PR TITLE
Add basic NameList to TOML file conversion

### PR DESCRIPTION
**Pull request by nefrathenrici**
# PULL REQUEST

## Purpose and Content
This PR enables conversion from NameList Dicts to TOML files as part of the larger parameter refactoring. It does not affect how NameList or TOML are used in the current model code.

## Benefits and Risks
In the future, this will be used to remove NameList usage and switch over to TOML, making for a more unified parameter file system. 

However, parsing a NameList Dict into a flat TOML file requires renaming some parameters, and I am not sure what kind of naming conventions to use. [ClimaParameters TOML docs](https://clima.github.io/CLIMAParameters.jl/dev/toml/) also state that there is only very limited support for different types, so this will need to be expanded.

At the moment there are no risks because this doesn't change how anything is run, but removing NameList later on might cause issues for some users. 

## Linked Issues
[ Refactor the TurbulenceConvection argument handling in ClimaAtmos #721 ](https://github.com/CliMA/ClimaAtmos.jl/issues/721)

## PR Checklist
- [x] This PR has a corresponding issue OR is linked to an SDI.
- [x] I have followed CliMA's codebase [contribution](https://clima.github.io/ClimateMachine.jl/latest/Contributing/) and [style](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) guidelines OR N/A.
- [x] I have followed CliMA's [documentation policy](https://github.com/CliMA/policies/wiki/Documentation-Policy).
- [x] I have checked all issues and PRs and I certify that this PR does not duplicate an open PR.
- [x] I linted my code on my local machine prior to submission OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code used in an integration test OR N/A.
- [x] All tests ran successfully on my local machine OR N/A.
- [x] All classes, modules, and function contain docstrings OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: tc_driver/generate_namelist.jl [hunk 1]
```diff
@@ -627,6 +627,88 @@ function write_file(namelist, root::String = ".")
     return
 end
 
+
```
[the remark this answers is not in the document]
Comment (trontrytel) [reply tc_driver/generate_namelist.jl:630]: I think they could stay here. I think they are kind of specific to how we used to do things here?

@odunbar any last veto on this?

**Review by odunbar — COMMENTED** [review]
LGTM 

I'm not sure people's opinion on this, but this PR (could/should) also contain
1. the created flat TOMLs from the namelists that you have converted
2. Unit test that match the values of the uploaded namelist and tomls.

**Comment by charleskawczynski** [comment]
> the created flat TOMLs from the namelists that you have converted

Can we make the toml files have only the overrides?

**Comment by nefrathenrici** [comment]
> Can we make the toml files have only the overrides?
This would actually be better, since I'm only currently testing the overrides and the TOML files should only have overrides in practice anyway. Thanks @charleskawczynski

**Comment by nefrathenrici** [comment]
bors try

File: Project.toml [hunk 2]
```diff
@@ -8,6 +8,7 @@ ClimaCore = "d414da3d-4745-48bb-8d80-42e94e092884"
 ClimaCorePlots = "cf7c7e5a-b407-4c48-9047-11a94a308626"
 ClimaCoreVTK = "c8b6d40d-e815-466f-95ae-c48aefa668fa"
 CloudMicrophysics = "6a9e3e04-43cd-43ba-94b9-e8782df3c71b"
+Colors = "5ae59095-9a9b-59fe-a467-6f913c188581"
```
Comment (charleskawczynski) [inline Project.toml:11]: I don't think we need to add this dep

File: Project.toml [hunk 3]
```diff
@@ -28,6 +29,7 @@ StaticArrays = "90137ffa-7385-5640-81b9-e52037218182"
 StatsBase = "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91"
 StochasticDiffEq = "789caeaf-c7a9-5a7d-9973-96adeb23e2a0"
 SurfaceFluxes = "49b00bb7-8bd4-4f2b-b78c-51cd0450215f"
+TOML = "fa267f1f-6049-4f14-aa54-33bafae1ed76"
```
Comment (charleskawczynski) [inline Project.toml:32]: I don't think we need to add this dep

Comment (charleskawczynski) [reply Project.toml:32]: Probably better to add to the other envs:
 - test/
 - perf/
 - examples/
 - docs/
This way ClimaAtmos.jl is not shipped with (nor depends on) it

**Comment by nefrathenrici** [comment]
bors try

**Comment by charleskawczynski** [comment]
Alright, I think I’m fine with this after squashing, good work, @nefrathenrici!

**Comment by nefrathenrici** [comment]
> Alright, I think I’m fine with this after squashing, good work, @nefrathenrici!

@charleskawczynski Do you mind taking a look at my rebase? I'm not sure I squashed everything correctly. [277d48c](https://github.com/CliMA/ClimaAtmos.jl/pull/812/commits/277d48c2e6ca9c68ae5128cf08e9ef848882b06c)

**Comment by nefrathenrici** [comment]
bors try

**Comment by charleskawczynski** [comment]
> > Alright, I think I’m fine with this after squashing, good work, @nefrathenrici!
> 
> @charleskawczynski Do you mind taking a look at my rebase? I'm not sure I squashed everything correctly. [277d48c](https://github.com/CliMA/ClimaAtmos.jl/pull/812/commits/277d48c2e6ca9c68ae5128cf08e9ef848882b06c)

Sure, I can help squash, I'll force push a squashed commit. Don't delete your local version!

**Comment by charleskawczynski** [comment]
There were a lot of conflicts, and I didn't want to mess things up so I pushed to a separate branch, [here](https://github.com/CliMA/ClimaAtmos.jl/compare/neh/ck/rebased/namelist_conversion), first. @nefrathenrici, let me know if it's alright to force-push here

**Comment by nefrathenrici** [comment]
@charleskawczynski Sounds good to me. Thanks for the help

**Review by charleskawczynski — APPROVED** [review]
Alright, @nefrathenrici, maybe just give it one final look over before r+ing?

I don't think it's a blocker, but the project hash change is a bit odd, I'm not sure what that's about. Any ideas @simonbyrne?

**Comment by charleskawczynski** [comment]
Oh, whoops, I forgot to re-apply the formatter. I'll do that and re-push now

**Comment by nefrathenrici** [comment]
LGTM. Should the project hash be reverted?

**Comment by nefrathenrici** [comment]
bors try

**Comment by charleskawczynski** [comment]
> LGTM. Should the project hash be reverted?

I think it's fine, maybe it's due to adding the toml file

**Pull request merged**